### PR TITLE
Added NoopInflector (null object pattern)

### PR DESCRIPTION
**Pull request by jasny**
The inflector does no operation (hence nop), simply returning the input.
This is an implementation of the Null object pattern, which prevents excessive use of `if` statements.

### CODE REVIEW

**Comment by jwage** [comment]
Cool. What is the most common spelling of something this? I always called it "noop". Thoughts @Majkl578 ?

**Comment by garrettw** [comment]
I've always seen it as "noop" too.

**Comment by Majkl578** [comment]
Noop. Any better suggestions?

**Comment by jwage** [comment]
Only other thought is NoOp. Some people say it like Noop, some say No Op.

**Comment by malarzm** [comment]
IIRC I've seen "nop" somewhere in ODM and on wiki it's also the main article (https://en.wikipedia.org/wiki/NOP) with "noop" redirecting there. My vote goes for "NopInflector" as PR already has :)

**Comment by Majkl578** [comment]
@malarzm That article is talking about Assembler instruction. :joy: 

Imho noop is more correct, as a shortand for _no operation_.
https://www.urbandictionary.com/define.php?term=no-op
https://whatis.techtarget.com/definition/no-op-no-operation
https://api.jquery.com/jquery.noop/
https://msdn.microsoft.com/cs-cz/library/s6btaxcs.aspx
https://www.python.org/dev/peps/pep-0559/
https://docs.angularjs.org/api/ng/function/angular.noop

**Comment by malarzm** [comment]
@Majkl578 not only Assembler instruction:

> [NOP] is an assembly language instruction, programming language statement, or computer protocol command that does nothing.

there are 3 uses mentioned ;)

**Comment by garrettw** [comment]
Just to further confuse the issue 😉, I just noticed that NOP also has another possible meaning that can be found in the title of this issue.

**Comment by jwage** [comment]
Lets go with `NoopInflector`. I think this is the most common and expected.

https://twitter.com/jwage/status/1026812207974240256

**Comment by jasny** [comment]
@jwage Changed to `NoopInflector` and added a section to documentation.

_P.S. Feel free to modify the PR if needed._

File: docs/en/index.rst [hunk 1]
```diff
@@ -194,6 +194,21 @@ If you want to setup custom singular and plural rules, you can configure the inf
         ))
     );
 
+No operation inflector
+----------------------
+
+The ``Doctrine\Common\Inflector\NoopInflector`` may be used to configure an inflector that doesn't perform any
```
Comment (Majkl578) [inline docs/en/index.rst]: Namespace is wrong here, master is no longer in Common.

File: docs/en/index.rst [hunk 2]
```diff
@@ -194,6 +194,21 @@ If you want to setup custom singular and plural rules, you can configure the inf
...
+operation for pluralization and/or singularization. If will simply return the input as output.
+
+This is an implementation of the `Null Object design pattern <https://sourcemaking.com/design_patterns/null_object>`_.
+
+.. code-block:: php
+
+    use Doctrine\Inflector\Inflector;
+    use Doctrine\Inflector\NoopInflector;
+
+    $inflector = new Inflector(new NoopInflector(), new NoopInflector());
```
Comment (Majkl578) [inline docs/en/index.rst]: This looks superconfusing. @jwage shouldn't it be NoopWordInflector instead?

Comment (jwage) [reply docs/en/index.rst]: I think I would agree. `NoopWordInflector` would be more consistent with https://github.com/doctrine/inflector/blob/master/lib/Doctrine/Inflector/CachedWordInflector.php :+1:

**Comment by jasny** [comment]
@jwage Guys, please just make the modifications and merge the PR instead of keep discussing it. You have the privileges to my repo to modify the PR.

**Comment by jwage** [comment]
Hi @jasny, if you want to see this functionality merged it would be helpful for you to see it all the way through. In open source projects, people usually work on what is important to them. It is likely this won't get merged if there is nobody from the outside sponsoring it and shepherding it through to merge. If you aren't interested in continuing the work, we can leave the PR open until someone else comes around that this is important to and wants to help finish it. Thanks for your contribution!

**Comment by jasny** [comment]
@jwage I also manage a number of open source projects. That's why I know how valuable a well written Pull Request is (including docs and tests).

It's not that I'm not willing to finish the PR, but I can't because you guys keep discussing the name, which is just silly. It really doesn't matter that much. I'm willing to make one final change, to get the PR merged. Otherwise I rather just close it and make a `jasny\doctrine-inflector-extension` library.

**Comment by jwage** [comment]
Updated the PR @jasny. Thanks for your contribution!

**Pull request merged**